Attempt to insert variable args array causes `builder error: cannot convert 'struct array' to 'struct string'`

### Describe the bug

Compilation of the below code fails with a `builder error: C error. This should never happen.`

```
fn  foo  ( args ...string )  {
  mut  v  :=  [ 'a' ]
  v .insert ( 1, [ 'b' ] )    //  this line compiles                            
  v .insert ( 1, args )       //  this line causes a 'builder error'            
}

fn  main  ()  {
  foo ( 'b' )
}
```


### Reproduction Steps

Try to compile the above code.

### Expected Behavior

The code will compile.

### Current Behavior

```
$  v -cg -o /tmp/test test.v
/tmp/v_1000/test.10658680212378167293.tmp.c:12678: error: cannot convert 'struct array' to 'struct string'
builder error: 
==================
C error. This should never happen.

This is a compiler bug, please report it using `v bug file.v`.

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang
```

### Possible Solution

None.

### Additional Information/Context

None.

### V version

V 0.4.3 17d540e

### Environment details (OS name and version, etc.)

Debian GNU/Linux 12.
> [!NOTE]
> You can use the 👍 reaction to increase the issue's priority for developers.
>
> Please note that only the 👍 reaction to the issue itself counts as a vote.
> Other reactions and those to comments will not be taken into account.

--- a/vlib/v/gen/c/array.v
+++ b/vlib/v/gen/c/array.v
@@ -825,7 +825,8 @@ fn (mut g Gen) gen_array_insert(node ast.CallExpr) {
 	left_info := left_sym.info as ast.Array
 	elem_type_str := g.typ(left_info.elem_type)
 	arg2_sym := g.table.sym(node.args[1].typ)
-	is_arg2_array := arg2_sym.kind == .array && node.args[1].typ == node.left_type
+	is_arg2_array := arg2_sym.kind == .array
+		&& node.args[1].typ.clear_flag(.variadic) == node.left_type
 	noscan := g.check_noscan(left_info.elem_type)
 	addr := if node.left_type.is_ptr() { '' } else { '&' }
 	if is_arg2_array {

--- a/vlib/v/tests/array_insert_variadic_arg_variable_test.v
+++ b/vlib/v/tests/array_insert_variadic_arg_variable_test.v
@@ -1,0 +1,12 @@
+fn foo(args ...string) []string {
+	mut v := ['a']
+	v.insert(1, ['b'])
+	v.insert(1, args)
+	return v
+}
+
+fn test_array_insert_variadic_arg_variable() {
+	ret := foo('b')
+	println(ret)
+	assert ret == ['a', 'b', 'b']
+}

